app/controllers/samples_controller.rb

```diff
@@ -952,7 +909,12 @@ def sample_taxons_dict(params)
                         else
                           JSON.parse(params[:thresholdFilters] || "[]")
                         end
-    include_phage = (JSON.parse(params[:subcategories]) || {}).fetch("Viruses", []).include?("Phage")
+    subcategories = if params[:subcategories].respond_to?(:to_h)
+                      params[:subcategories].permit!.to_h
```

Comment:
Sorry, do not know enough about this... what is this branch doing ?

Reply:
I got it... i guess now the parameter is not necessary a string, and we were allowing all keys already.

Reply:
exactly